<!-- Main launcher the Lane Detector

Includes:
  -Camera (containing nodelet manager and processing nodes)
  -Lane detector nodelet
     -Subscribes to image_rect_mono
-->
<launch>
  
  <arg name="camera_name"  value="camera" />
  <arg name="manager"      value="camera_nodelet_manager" />
  
  <!-- Camera -->
  <include file="$(find ghost)/launch/camera.launch" >
    <arg name="manager"      value="$(arg manager)" />
    <arg name="camera_name"  value="$(arg camera_name)" />
  </include>
    
  <!-- Lane Detector -->
  <node 
    pkg="nodelet" 
    type="nodelet" 
    name="lane_detector" 
    args="load ghost/LaneDetector $(arg camera_name)/$(arg manager)" 
    output="screen">
    <remap from="/$(arg camera_name)/image_rect_mono" to="$(arg camera_name)/cropped/image_rect_mono"/>
    
    <!-- General detection parameters -->
    <param name="debug" value="false" />
    <param name="min_range" value="0.30" />
    <param name="max_range" value="4.0" />
    <param name="detect_dx" value="0.10" />
    <param name="detect_frame_id" value="camera_ground" />
    <param name="imu_topic" value="/imu/data" />
    <param name="use_imu" value="false" />
    <param name="bev_ppm" value="200" />
    
    <!-- Image filtering parameters-->
    <param name="apply_gauss" value="true" />
    <param name="gauss_k_size" value="5" />
    <param name="gauss_sigma_x" value="0.8" />
    <param name="gauss_sigma_y" value="0.4" />
    <param name="canny_threshold" value="0" />
    <param name="canny_lower" value="120" />
    <param name="canny_upper" value="150" />
    <param name="scan_whole_row" value="false" />
    <param name="lane_centre_alpha" value="0.95" />
  </node>

</launch>